### ISSUE TITLE
Implement Train / Validation / Test

### ISSUE THREAD

**Issue by comane** [HEP-PBSP/colibri#11]
Need test set for correct hyper optimisation.

Note: 
the main problem lies in the fact that we are working with correlated datapoints. Hence, when performing a random train / test split, we might end up with train and test sets correlated between each others. This, in particular, might result in a too optimistic loss function on test set.

A possible solution might be the following:
-  Diagonalise the covariance matrix (and project datapoints vectors in the corresponding space). 
- Perform Train / Test split  and generate replicas
- Perform fit using the diagonalised covariance matrix

Problem:
pseudo data (replica) sampling is done with the experimental covmat
fit is performed with t0 covmat -> the change of basis is not the same for pseudo data generation and fit.
